frontend/src/services/ApiService.ts

```diff
@@ -78,3 +78,32 @@ ApiService.interceptors.response.use(
     throw transformedError
   },
 )
+
+export const processFetchResponse = async (response: Response) => {
+  try {
+    const data = response.json()
+
+    // TODO: data may be present but if response is non 2XX, transform to error and throw
+    return data
+  } catch (error) {
+    datadogLogs.logger.warn(`Fetch error: ${error.message}`, {
+      meta: {
+        action: 'processFetchResponse',
+        response: {
+          status: response.status,
+          statusText: response.statusText,
+          headers: [...(response.headers?.entries() || [])],
+          body: response.body,
+        },
+        error: {
+          code: error?.code,
```

Comment:
from what I can tell, there's no guarantee that the error will contain the `code` property - so omitted this.

Reply:
It would have been fine to leave it I think. As much as we want to have type safety everywhere, if we do get errors with unexpected shape and a code field, making our extraction grab it works (in contrast to json encoding of the error object which will skip all non-enumerable properties). Plus If `error.code` is `undefined`, json stringification will remove the field anyway when it is sent to DD.

Reply:
[a52c228](https://github.com/opengovsg/FormSG/pull/6008/commits/a52c228ca54df85ff215ac97c76f91bdaed84e39)